test/parallel/test-fs-read-stream-inherit.js

```diff
@@ -7,174 +7,189 @@ const fs = require('fs');
 const fn = path.join(common.fixturesDir, 'elipses.txt');
 const rangeFile = path.join(common.fixturesDir, 'x.txt');
 
-const callbacks = { open: 0, end: 0, close: 0 };
-
 let paused = false;
 
-const file = fs.ReadStream(fn);
+{
+  const file = fs.ReadStream(fn);
 
-file.on('open', function(fd) {
-  file.length = 0;
-  callbacks.open++;
-  assert.strictEqual(typeof fd, 'number');
-  assert.ok(file.readable);
+  file.on('open', common.mustCall(function(fd) {
+    file.length = 0;
+    assert.strictEqual(typeof fd, 'number');
+    assert.ok(file.readable);
 
-  // GH-535
-  file.pause();
-  file.resume();
-  file.pause();
-  file.resume();
-});
+    // GH-535
+    file.pause();
+    file.resume();
+    file.pause();
+    file.resume();
+  }));
 
-file.on('data', function(data) {
-  assert.ok(data instanceof Buffer);
-  assert.ok(!paused);
-  file.length += data.length;
+  file.on('data', function(data) {
+    assert.ok(data instanceof Buffer);
+    assert.ok(!paused);
+    file.length += data.length;
 
-  paused = true;
-  file.pause();
+    paused = true;
+    file.pause();
 
-  setTimeout(function() {
-    paused = false;
-    file.resume();
-  }, 10);
-});
+    setTimeout(function() {
+      paused = false;
+      file.resume();
+    }, 10);
+  });
 
 
-file.on('end', function(chunk) {
-  callbacks.end++;
-});
+  file.on('end', common.mustCall(function() {}));
 
 
-file.on('close', function() {
-  callbacks.close++;
-});
+  file.on('close', common.mustCall(function() {}));
 
-const file3 = fs.createReadStream(fn, Object.create({encoding: 'utf8'}));
-file3.length = 0;
-file3.on('data', function(data) {
-  assert.strictEqual(typeof data, 'string');
-  file3.length += data.length;
+  process.on('exit', function() {
+    assert.strictEqual(file.length, 30000);
```

Comment:
I suppose this can be checked in `file.on('close')` like what's being done for file4.